provider/nomad/nomad_discover.go

```diff
@@ -0,0 +1,155 @@
+// Package nomad provides allocation discovery for Nomad.
+package nomad
+
+import (
+	"fmt"
+	"log"
+
+	"encoding/base64"
+
+	"github.com/hashicorp/nomad/api"
+)
+
+type Provider struct{
+	// client *api.Client
+}
+
+func (p *Provider) Help() string {
+	return `Nomad (Nomad):
+
+    provider:         "nomad"
+		address:    			Nomad address
+		secret_id:    	  Nomad secret_id
+    service_name:     Nomad service to discover allocations for
+    namespace:        Namespace to search for allocations (optional)
+		region:     			Nomad region to discover allocations for (optional)
+`
+}
+
+func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error) {
+	if args["provider"] != "nomad" {
+		return nil, fmt.Errorf("discover-nomad: invalid provider " + args["provider"])
+	}
+
+	serviceName := args["service_name"]
+	if serviceName == "" {
+		return nil, fmt.Errorf("discover-nomad: must provider a service_name")
+	}
+
+	client, err := createNomadClient(args)
+	if err != nil {
+		return nil, fmt.Errorf("discover-nomad: error creating client: %v", err)
+	}
+
+	// TODO: Do I need NS and Region here if provided above?
+	queryOpts := &api.QueryOptions{}
+	services, _, err := client.Services().Get(serviceName, queryOpts)
+	if err != nil {
+		return nil, fmt.Errorf("discover-nomad: error retrieving services: %v", err)
+	}
+
+	addrs := []string{}
+	for _, service := range services {
+		addr := service.Address
+		port := service.Port
+		addrs = append(addrs, fmt.Sprintf("%s:%d", addr, port))
+	}
+
+	return addrs, nil
+}
+
+func createNomadClient(args map[string]string) (*api.Client, error) {
+	clientConfig := api.DefaultConfig()
+
+	// === BASIC CONFIG ===
+
+	if args["address"] == "" {
+		return nil, fmt.Errorf("discover-nomad: must provider an address for Nomad")
+	} else {
+		clientConfig.Address = args["address"]
+	}
+	if args["namespace"] != "" {
+		clientConfig.Namespace = args["namespace"]
+	}
+	if args["region"] != "" {
+		clientConfig.Region = args["region"]
+	}
+	if args["secret_id"] != "" {
```

Comment:
Does something need to be done to pass the WI token, if present, here or is there hidden magic?

Reply:
This is available as an ENV, as is the unix socket to talk to Nomad, so it should just be an issue of passing in the env var as the secret ID properly.